rust/ql/src/queries/unusedentities/UnusedValue.ql

```diff
@@ -9,7 +9,16 @@
  */
 
 import rust
+import codeql.rust.dataflow.Ssa
+import codeql.rust.dataflow.internal.SsaImpl
+import UnusedVariable
 
-from Locatable e
-where none() // TODO: implement query
-select e, "Variable is assigned a value that is never used."
+from AstNode write, Ssa::Variable v
+where
+  variableWrite(write, v) and
+  // SSA definitions are only created for live writes
```

Comment:
What is a "live write"?

Reply:
One that can reach a read.

Reply:
Ah of course, the key line of the query.